test/jb_directory/add_known_controller.test.js

```diff
@@ -0,0 +1,64 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+
+import { deployMockContract } from '@ethereum-waffle/mock-contract';
+
+import jbController from '../../artifacts/contracts/interfaces/IJBController.sol/IJBController.json';
+import jbOperatoreStore from '../../artifacts/contracts/JBOperatorStore.sol/JBOperatorStore.json';
+import jbProjects from '../../artifacts/contracts/JBProjects.sol/JBProjects.json';
+
+describe('JBDirectory::addKnownController(...)', function () {
+
+  async function setup() {
+    let [deployer, ...addrs] = await ethers.getSigners();
+
+    let mockJbOperatorStore = await deployMockContract(deployer, jbOperatoreStore.abi);
+    let mockJbProjects = await deployMockContract(deployer, jbProjects.abi);
+    let mockJbController = await deployMockContract(deployer, jbController.abi);
+
+    let jbDirectoryFactory = await ethers.getContractFactory('JBDirectory');
+    let jbDirectory = await jbDirectoryFactory.deploy(
+      mockJbOperatorStore.address,
+      mockJbProjects.address,
+    );
+
+    return {
+      deployer,
+      addrs,
+      jbDirectory,
+      mockJbController
+    };
+  }
+
+  it('Should add known controller and emit events if caller is JBDirectory owner', async function () {
+    const { deployer, jbDirectory, mockJbController } = await setup();
+
+    await expect(
+      jbDirectory.connect(deployer).addToSetControllerAllowlist(mockJbController.address)
+    ).to.emit(jbDirectory, 'AddToSetControllerAllowlist')
+      .withArgs(mockJbController.address, deployer.address);
+
+    expect(
+      await jbDirectory.isAllowedToSetController(mockJbController.address)
+    ).to.be.true;
+  });
+
+  it('Can\'t add known controller if caller is not JBDirectory owner', async function () {
```

Comment:
If \ characters are in the name, you will need to wrap with ` ` quotes - otherwise the test name will show as, "Can"

Reply:
Fix in all files in this PR.

Reply:
the \ is to escape the ' character, display is ok but double-checking other tests, this is indeed not consistent across them

![image](https://user-images.githubusercontent.com/83670532/145053098-4c4f8834-aa0d-4280-b6f2-193bf88d1f41.png)